### PR TITLE
refactor: metric number formatting and add unit tests

### PR DESCRIPTION
**Pull request by 0xAlunara**
* Refactor the Metric component by extracting number formatting and unit handling into utility functions.
* Introduce Vitest for testing and add unit tests for number formatting, ensuring consistent behavior across edge cases.
* Update documentation and fix minor issues related to number formatting.

This PR does not change anything visibly. This is the pre-work required to replace the number formatting from `utilsFormat.ts`, so that we can consolidate the number formatting to use the same logic everywhere.

### CODE REVIEW

File: packages/curve-ui-kit/src/utils/number.spec.ts [hunk 1]
```diff
@@ -0,0 +1,568 @@
...
+      expect(formatNumber(1234567.89, { useGrouping: true, abbreviate: false })).toBe('1,234,567.89')
+      expect(formatNumber(12.5, { decimals: 4, trailingZeroDisplay: 'stripIfInteger', abbreviate: false })).toBe(
+        '12.5000',
+      )
+      expect(formatNumber(12.0, { decimals: 4, trailingZeroDisplay: 'stripIfInteger', abbreviate: false })).toBe('12')
+      expect(formatNumber(12.5, { decimals: 4, trailingZeroDisplay: 'auto', abbreviate: false })).toBe('12.5000')
+    })
+
+    it('handles JSDoc examples with custom formatter', () => {
+      const customFormatter = (value: number) => `[${value}]`
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/utils/number.spec.ts:564]: Holy crap that's a lot of cases

**Review by DanielSchiavini — APPROVED** [review]

**Review by DanielSchiavini — APPROVED** [review]

**Review by DanielSchiavini — APPROVED** [review]

**Review by OnlyJousting — APPROVED** [review]
